### DISCUSSION TITLE
HPA scaling event (likely scaling down event) seems to affect analysisruns

### DISCUSSION THREAD

**Discussion by schlags** [argoproj/argo-rollouts#2935] [Q&A]
We're noticing some odd behavior in our pre/post-analysisruns for BlueGreen. 
It seems to be a relation where if the HPA scales down the desired pod count in the RS, the reconciliation of pre/post analysis will either terminate the running analysisrun and continue with the rollout, or it will see a completed analysisrun and throw a "Collision" log. Eventually spinning up a `.1` and sometimes a `.2` AnalysisRun.

The former ("Run Terminated" with Success status) is very concerning since it allows our deployment to continue without completing analysis. 

I haven't been able to find any conclusive documentation around how this can be mitigated. Any thoughts?

**Comment by gavin-db** [comment]
Allegedly duplicate: https://github.com/argoproj/argo-rollouts/issues/2938#issuecomment-1736075416
